### PR TITLE
Volume potential check

### PR DESCRIPTION
**Pull request by simonOpenV**

### CODE REVIEW

File: ovs/extensions/healthcheck/volumedriver_hc.py [hunk 1]
```diff
@@ -327,3 +327,32 @@ def check_filedrivers(result_handler):
...
+
+        for std in VolumedriverHealthCheck.LOCAL_SR.storagedrivers:
+            std_driver_log_string = 'Volume potential of local storage driver {0}'.format(std.storagedriver_id)
+            try:
+                std_config = StorageDriverConfiguration(std.vpool.guid, std.storagedriver_id)
+                client = src.LocalStorageRouterClient(std_config.remote_path)
+                vol_potential = client.volume_potential(str(std.storagedriver_id))
+                if vol_potential >= critical_volume_number:
+                    result_handler.success('{0} exceeds critical volume limit ({1}>>>{2})'.format(std_driver_log_string, vol_potential, critical_volume_number))
+                elif critical_volume_number>vol_potential>0:
```
Comment (kvanhijf) [inline ovs/extensions/healthcheck/volumedriver_hc.py]: Spacing

File: ovs/extensions/healthcheck/volumedriver_hc.py [hunk 2]
```diff
@@ -327,3 +327,32 @@ def check_filedrivers(result_handler):
...
+        import volumedriver.storagerouter.storagerouterclient as src
+
+        for std in VolumedriverHealthCheck.LOCAL_SR.storagedrivers:
+            std_driver_log_string = 'Volume potential of local storage driver {0}'.format(std.storagedriver_id)
+            try:
+                std_config = StorageDriverConfiguration(std.vpool.guid, std.storagedriver_id)
+                client = src.LocalStorageRouterClient(std_config.remote_path)
+                vol_potential = client.volume_potential(str(std.storagedriver_id))
+                if vol_potential >= critical_volume_number:
+                    result_handler.success('{0} exceeds critical volume limit ({1}>>>{2})'.format(std_driver_log_string, vol_potential, critical_volume_number))
```
Comment (kvanhijf) [inline ovs/extensions/healthcheck/volumedriver_hc.py]: Kinda strange message when everything is ok
When exceeding a critical volume limit, i'd presume something is wrong

File: ovs/extensions/healthcheck/volumedriver_hc.py [hunk 3]
```diff
@@ -327,3 +327,32 @@ def check_filedrivers(result_handler):
...
+                elif critical_volume_number>vol_potential>0:
+                    result_handler.warning('{0} is critically low ({1} volumes left)'.format(std_driver_log_string, vol_potential))
+                elif critical_volume_number <=0:
```
Comment (kvanhijf) [inline ovs/extensions/healthcheck/volumedriver_hc.py]: Spacing

File: ovs/extensions/healthcheck/volumedriver_hc.py [hunk 4]
```diff
@@ -327,3 +327,32 @@ def check_filedrivers(result_handler):
...
+                    result_handler.failure('{0} has no more spare volumes'.format(std_driver_log_string))
+            except RuntimeError:
```
Comment (kvanhijf) [inline ovs/extensions/healthcheck/volumedriver_hc.py]: Only put the stuff that can throw error in `except` clause

File: ovs/extensions/healthcheck/volumedriver_hc.py [hunk 5]
```diff
@@ -327,3 +327,32 @@ def check_filedrivers(result_handler):
...
+            try:
+                std_config = StorageDriverConfiguration(std.vpool.guid, std.storagedriver_id)
+                client = src.LocalStorageRouterClient(std_config.remote_path)
+                vol_potential = client.volume_potential(str(std.storagedriver_id))
+                if vol_potential >= critical_volume_number:
+                    result_handler.success('{0} exceeds critical volume limit ({1}>>>{2})'.format(std_driver_log_string, vol_potential, critical_volume_number))
+                elif critical_volume_number>vol_potential>0:
+                    result_handler.warning('{0} is critically low ({1} volumes left)'.format(std_driver_log_string, vol_potential))
+                elif critical_volume_number <=0:
+                    result_handler.failure('{0} has no more spare volumes'.format(std_driver_log_string))
```
Comment (kvanhijf) [inline ovs/extensions/healthcheck/volumedriver_hc.py]: Strange sentence when formatting complete

File: ovs/extensions/healthcheck/volumedriver_hc.py [hunk 6]
```diff
@@ -327,3 +325,29 @@ def check_filedrivers(result_handler):
...
+    def check_volume_potential(result_handler, critical_vol_number=25):
+        """
+        Checks all local storage drivers from a volume driver. Results in a success if enough volumes are available, a warning if the number of volumes is
+        lower then a threshold value (critical_volume_number) and a failure if the nr of volumes ==0)
+        :param result_handler: logging object
+        :type result_handler: ovs.extensions.healthcheck.result.HCResults
+        :param critical_vol_number: maximal number of volumes that result in a warning
+        :type critical_vol_number: int
+        """
+        for std in VolumedriverHealthCheck.LOCAL_SR.storagedrivers:
```
Comment (kvanhijf) [inline ovs/extensions/healthcheck/volumedriver_hc.py]: Check if `critical_vol_number` is integer

File: ovs/extensions/healthcheck/volumedriver_hc.py [hunk 7]
```diff
@@ -327,3 +325,29 @@ def check_filedrivers(result_handler):
...
+            try:
+                std_config = StorageDriverConfiguration(std.vpool.guid, std.storagedriver_id)
```
Comment (kvanhijf) [inline ovs/extensions/healthcheck/volumedriver_hc.py]: Use `std.vpool_guid` instead

File: ovs/extensions/healthcheck/volumedriver_hc.py [hunk 8]
```diff
@@ -327,3 +325,29 @@ def check_filedrivers(result_handler):
...
+                vol_potential = client.volume_potential(str(std.storagedriver_id))
+                if vol_potential >= critical_vol_number:
+                    log_level = 'success'
+                elif critical_vol_number > vol_potential > 0:
+                    log_level = 'warning'
+                else:
+                    log_level = 'failure'
+                getattr(result_handler, log_level)('Volume potential of local storage driver: {0}: {1} (potential at: {2})'.format(std.storagedriver_id, log_level.upper(), vol_potential))
+            except RuntimeError:
+                result_handler.exception('Unable to retrieve {0}'.format(std.storagedriver_id.lower()))
```
Comment (kvanhijf) [inline ovs/extensions/healthcheck/volumedriver_hc.py]: Unclear exception message and no need to cast an ID to lowercase

File: ovs/extensions/healthcheck/volumedriver_hc.py [hunk 9]
```diff
@@ -327,3 +325,32 @@ def check_filedrivers(result_handler):
...
+    def check_volume_potential(result_handler, critical_vol_number=25):
+        """
+        Checks all local storage drivers from a volume driver. Results in a success if enough volumes are available, a warning if the number of volumes is
+        lower then a threshold value (critical_volume_number) and a failure if the nr of volumes ==0)
+        :param result_handler: logging object
+        :type result_handler: ovs.extensions.healthcheck.result.HCResults
+        :param critical_vol_number: maximal number of volumes that result in a warning
+        :type critical_vol_number: int
+        """
+        if not isinstance(critical_vol_number, int):
```
Comment (JeffreyDevloo) [inline ovs/extensions/healthcheck/volumedriver_hc.py]: Well if you want to do input checking, why not check only positive integers ;)

**Pull request merged**